src/internal/hooks/use-performance-marks/index.ts

```diff
@@ -1,13 +1,40 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-import { useEffect } from 'react';
+import { useEffect, useState } from 'react';
 
 import { useModalContext } from '../../context/modal-context';
 import { useDOMAttribute } from '../use-dom-attribute';
 import { useEffectOnUpdate } from '../use-effect-on-update';
 import { useRandomId } from '../use-unique-id';
 
+const EVALUATE_COMPONENT_VISIBILITY_EVENT = 'awsui-evaluate-component-visibility';
+
+/**
+ * This hook manages a boolean state (`evaluateComponentVisibility`) that toggles
+ * whenever a custom DOM event (`EVALUATE_COMPONENT_VISIBILITY_EVENT`) is triggered.
+ *
+ * (`EVALUATE_COMPONENT_VISIBILITY_EVENT`) event is removed when the component unmounts
```

Comment:
An event can't be removed, we're referring to the event _listener_ here. I would recommend to either just remove the line (preferred) because it is not relevant to the consumer of this function, or alternatively extend it (see suggestion below)



```suggestion
```

```suggestion
 * The `EVALUATE_COMPONENT_VISIBILITY_EVENT` event listener is removed when the component unmounts.
```